app/services/toolbox/dalle.rb

```diff
@@ -0,0 +1,19 @@
+class Toolbox::Dalle < Toolbox
+
+  describe :generate_an_image, <<~S
+    Generate an image based on what the user asks you to generate. You will pass the user's prompt and will get back a URL to an image.
+  S
+
+  def self.generate_an_image(image_generation_prompt_s:)
+
+    client = client = OpenAI::Client.new(
+      access_token: User.last.openai_key,
+    )
+
+    response = client.images.generate(parameters: { prompt: image_generation_prompt_s, model: "dall-e-3", size: "1024x1792", quality: "standard" })
+
+    {
+      url: response.dig("data", 0, "url")
+    }
+  end
```

Comment:
My one small stylistic suggestion is to put that client into a private helper `def client`